vitrivr-engine-query/src/main/kotlin/org/vitrivr/engine/query/parsing/QueryParser.kt

```diff
@@ -83,7 +91,47 @@ class QueryParser(val schema: Schema) {
                 field.getRetrieverForDescriptor(descriptor, description.context)
             }
             is VectorInputData -> field.getRetrieverForDescriptor(FloatVectorDescriptor(vector = input.data.map { Value.Float(it) }, transient = true), description.context)
-            else -> field.getRetrieverForContent(content.computeIfAbsent(operation.input) { input.toContent() }, description.context)
+            else -> {
+                /* Is this a boolean sub-field query ? */
+                if(input.comparison != null){
```

Comment:
Shouldn't this be `if(fieldAndAttributeName.second != null)` then?

Reply:
Yeah, I guess this should be a bit more careful regarding the possible cases. Currently, a Boolean query on a sub-field *requires* both, a sub-field address (via the dot-notation) *and* a comparison operator.